File: apache-spark/src/main/scala/com/cognologix/scala/spark/WordCount.scala
package com.cognologix.scala.spark

import org.apache.log4j.{Level, Logger}
import org.apache.spark.{SparkConf, SparkContext}

object WordCount {

  def main(args: Array[String]): Unit = {

    Logger.getLogger("org").setLevel(Level.OFF)
    Logger.getLogger("akka").setLevel(Level.OFF)

    val sparkConf = new SparkConf().setAppName("First Spark Program").setMaster("local[*]")

    val sparkContext = new SparkContext(sparkConf)

    val rdd = sparkContext.textFile("apache-spark/in/word_count.text")

    /*
      calculate number of words from text file
      and print it on the console.
     */
    
    // count total number of words 
    
    val result = rdd.map(_.split(" ").length)
    print(result.sum().toInt)
    
    /* Another way to count total number of words 
    val words = rdd.flatMap(x => x.split(" "))
    println(words.count())
    */
   
    // occurrence 
    val words = rdd.flatMap(x => x.split(" "))
    val wordCount = words.countByValue()
    wordCounts.foreach(println)
   
    sparkContext.stop()
  }

}

File: apache-spark/src/main/java/com/cognologix/spark/WordCount.java
package com.cognologix.spark;

/*
      calculate number of words from text file
      and print it on the console.
*/

import org.apache.log4j.Level;
import org.apache.log4j.Logger;
import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.FlatMapFunction;

import java.util.Arrays;
import java.util.Iterator;

public class WordCount {

    public static void main(String[] args) {

        SparkConf conf = new SparkConf().setMaster("local[*]").setAppName("AverageHousePrice");
        JavaSparkContext sc = new JavaSparkContext(conf);

        Logger.getLogger("org.apache").setLevel(Level.OFF);

        JavaRDD<String> ip = sc.textFile("apache-spark/in/word_count.text").flatMap(new FlatMapFunction<String, String>() {
            @Override
            public Iterator<String> call(String s) throws Exception {
                return Arrays.stream(s.split(" ")).iterator();
            }
        });

        System.out.println(ip.count());

    }
}
